control-plane/agents/src/bin/core/controller/scheduling/volume.rs

```diff
@@ -80,6 +87,9 @@ impl GetSuitablePoolsContext {
     pub(crate) fn move_repl(&self) -> Option<&MoveReplica> {
         self.move_repl.as_ref()
     }
+    pub(crate) fn vg_restricted_nodes(&self) -> &Option<HashSet<NodeId>> {
```

Comment:
The number of nodes is usually not very large, so a vector probably fits the bill better here?

Reply:
Used HashSet to prevent duplicates. Operations are pretty much similar though.